pandas/_libs/lib.pyx

```diff
@@ -1223,6 +1223,28 @@ try:
 except AttributeError:
     pass
 
+cdef enum CurrSeen:
+    bool_ = 0
+    null_ = 1
```

Comment:
does null_here distinguish between nan/pd.NA/...?

Reply:
Nope, it shouldn't matter anyways.

The final inferred dtype will determine the type of the na-values.